<?xml version="1.0" encoding="UTF-8" ?>

<testsuite failures="0" time="2.349" errors="7" skipped="0" tests="2" name="ApplicationTest">
  <testcase classname="ApplicationTest" name="loginUser" time="0.385">
                        <error type="" message="A java.lang.RuntimeException has been caught, java.util.concurrent.ExecutionException: play.exceptions.TemplateNotFoundException: Template not found : main.html">
        <![CDATA[
          In /test/ApplicationTest.java, line 100
          
              	    GET(activateUrl+&quot;/&quot; +  user.validationCode); :
          
        ]]>
        </error>
                  <system-err>
            java.lang.RuntimeException: java.util.concurrent.ExecutionException: play.exceptions.TemplateNotFoundException: Template not found : main.html
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:299)
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:305)
	at play.test.FunctionalTest.GET(FunctionalTest.java:103)
	at play.test.FunctionalTest.GET(FunctionalTest.java:57)
	at ApplicationTest.loginUser(ApplicationTest.java:100)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at play.test.PlayJUnitRunner$StartPlay$2$1.evaluate(PlayJUnitRunner.java:105)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:76)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at play.test.PlayJUnitRunner.run(PlayJUnitRunner.java:55)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:24)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:157)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:117)
	at play.test.TestEngine.run(TestEngine.java:112)
	at controllers.TestRunner.run(TestRunner.java:66)
	at sun.reflect.GeneratedMethodAccessor50.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
	at play.server.PlayHandler$NettyInvocation.execute(PlayHandler.java:257)
	at play.Invoker$Invocation.run(Invoker.java:278)
	at play.server.PlayHandler$NettyInvocation.run(PlayHandler.java:235)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:98)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:206)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
	at java.lang.Thread.run(Thread.java:680)
Caused by: java.util.concurrent.ExecutionException: play.exceptions.TemplateNotFoundException: Template not found : main.html
	at java.util.concurrent.FutureTask$Sync.innerGet(FutureTask.java:232)
	at java.util.concurrent.FutureTask.get(FutureTask.java:91)
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:286)
	... 56 more
Caused by: play.exceptions.TemplateNotFoundException: Template not found : main.html
	at play.templates.FastTags._extends(FastTags.java:335)
	at play.templates.FastTags$_extends.call(Unknown Source)
	at /app/views/Application/activate.html.(line:1)
	at play.templates.GroovyTemplate.internalRender(GroovyTemplate.java:232)
	at play.templates.Template.render(Template.java:26)
	at play.templates.GroovyTemplate.render(GroovyTemplate.java:187)
	at play.mvc.results.RenderTemplate.&lt;init&gt;(RenderTemplate.java:24)
	at play.mvc.Controller.renderTemplate(Controller.java:660)
	at play.mvc.Controller.renderTemplate(Controller.java:640)
	at play.mvc.Controller.render(Controller.java:695)
	at controllers.Application.activate(Application.java:145)
	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
	at play.test.FunctionalTest$1.execute(FunctionalTest.java:269)
	at play.Invoker$Invocation.run(Invoker.java:278)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	... 3 more

          </system-err>  
        }
            </testcase>
  <testcase classname="ApplicationTest" name="testThatIndexPageWorks" time="0.754">
      </testcase>
  <testcase classname="ApplicationTest" name="getDealListForIphone" time="0.282">
                        <error type="" message="A java.lang.RuntimeException has been caught, java.util.concurrent.ExecutionException: play.exceptions.JavaExecutionException">
        <![CDATA[
          In /test/ApplicationTest.java, line 72
          
              		Response response = GET(&quot;/deals/madrid&quot;); :
          
        ]]>
        </error>
                  <system-err>
            java.lang.RuntimeException: java.util.concurrent.ExecutionException: play.exceptions.JavaExecutionException
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:299)
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:305)
	at play.test.FunctionalTest.GET(FunctionalTest.java:103)
	at play.test.FunctionalTest.GET(FunctionalTest.java:57)
	at ApplicationTest.getDealListForIphone(ApplicationTest.java:72)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at play.test.PlayJUnitRunner$StartPlay$2$1.evaluate(PlayJUnitRunner.java:105)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:76)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at play.test.PlayJUnitRunner.run(PlayJUnitRunner.java:55)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:24)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:157)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:117)
	at play.test.TestEngine.run(TestEngine.java:112)
	at controllers.TestRunner.run(TestRunner.java:66)
	at sun.reflect.GeneratedMethodAccessor50.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
	at play.server.PlayHandler$NettyInvocation.execute(PlayHandler.java:257)
	at play.Invoker$Invocation.run(Invoker.java:278)
	at play.server.PlayHandler$NettyInvocation.run(PlayHandler.java:235)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:98)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:206)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
	at java.lang.Thread.run(Thread.java:680)
Caused by: java.util.concurrent.ExecutionException: play.exceptions.JavaExecutionException
	at java.util.concurrent.FutureTask$Sync.innerGet(FutureTask.java:232)
	at java.util.concurrent.FutureTask.get(FutureTask.java:91)
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:286)
	... 56 more
Caused by: play.exceptions.JavaExecutionException
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:231)
	at play.test.FunctionalTest$1.execute(FunctionalTest.java:269)
	at play.Invoker$Invocation.run(Invoker.java:278)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	... 3 more
Caused by: java.lang.NullPointerException
	at controllers.Deals.iPhoneList(Deals.java:154)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
	... 8 more

          </system-err>  
        }
            </testcase>
  <testcase classname="ApplicationTest" name="validateUser" time="0.149">
                        <error type="" message="A java.lang.RuntimeException has been caught, java.util.concurrent.ExecutionException: play.exceptions.TemplateNotFoundException: Template not found : main.html">
        <![CDATA[
          In /test/ApplicationTest.java, line 86
          
              	    GET(activateUrl+&quot;/&quot; +  user.validationCode); :
          
        ]]>
        </error>
                  <system-err>
            java.lang.RuntimeException: java.util.concurrent.ExecutionException: play.exceptions.TemplateNotFoundException: Template not found : main.html
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:299)
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:305)
	at play.test.FunctionalTest.GET(FunctionalTest.java:103)
	at play.test.FunctionalTest.GET(FunctionalTest.java:57)
	at ApplicationTest.validateUser(ApplicationTest.java:86)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at play.test.PlayJUnitRunner$StartPlay$2$1.evaluate(PlayJUnitRunner.java:105)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:76)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at play.test.PlayJUnitRunner.run(PlayJUnitRunner.java:55)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:24)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:157)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:117)
	at play.test.TestEngine.run(TestEngine.java:112)
	at controllers.TestRunner.run(TestRunner.java:66)
	at sun.reflect.GeneratedMethodAccessor50.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
	at play.server.PlayHandler$NettyInvocation.execute(PlayHandler.java:257)
	at play.Invoker$Invocation.run(Invoker.java:278)
	at play.server.PlayHandler$NettyInvocation.run(PlayHandler.java:235)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:98)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:206)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
	at java.lang.Thread.run(Thread.java:680)
Caused by: java.util.concurrent.ExecutionException: play.exceptions.TemplateNotFoundException: Template not found : main.html
	at java.util.concurrent.FutureTask$Sync.innerGet(FutureTask.java:232)
	at java.util.concurrent.FutureTask.get(FutureTask.java:91)
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:286)
	... 56 more
Caused by: play.exceptions.TemplateNotFoundException: Template not found : main.html
	at play.templates.FastTags._extends(FastTags.java:335)
	at play.templates.FastTags$_extends.call(Unknown Source)
	at /app/views/Application/activate.html.(line:1)
	at play.templates.GroovyTemplate.internalRender(GroovyTemplate.java:232)
	at play.templates.Template.render(Template.java:26)
	at play.templates.GroovyTemplate.render(GroovyTemplate.java:187)
	at play.mvc.results.RenderTemplate.&lt;init&gt;(RenderTemplate.java:24)
	at play.mvc.Controller.renderTemplate(Controller.java:660)
	at play.mvc.Controller.renderTemplate(Controller.java:640)
	at play.mvc.Controller.render(Controller.java:695)
	at controllers.Application.activate(Application.java:145)
	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
	at play.test.FunctionalTest$1.execute(FunctionalTest.java:269)
	at play.Invoker$Invocation.run(Invoker.java:278)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	... 3 more

          </system-err>  
        }
            </testcase>
  <testcase classname="ApplicationTest" name="createUserFromJson" time="0.309">
                        <error type="" message="A java.lang.NullPointerException has been caught, Try to read content on null object models.dto.UserStatusMessage (ApplicationTest.createUserFromJson, line 129)">
        <![CDATA[
          In /test/ApplicationTest.java, line 129
          
              	    UserDTO newUser =  message.content; :
          
        ]]>
        </error>
                  <system-err>
            java.lang.NullPointerException: Try to read content on null object models.dto.UserStatusMessage (ApplicationTest.createUserFromJson, line 129)
	at play.classloading.enhancers.PropertiesEnhancer$FieldAccessor.invokeReadProperty(PropertiesEnhancer.java:205)
	at ApplicationTest.createUserFromJson(ApplicationTest.java:129)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at play.test.PlayJUnitRunner$StartPlay$2$1.evaluate(PlayJUnitRunner.java:105)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:76)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at play.test.PlayJUnitRunner.run(PlayJUnitRunner.java:55)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:24)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:157)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:117)
	at play.test.TestEngine.run(TestEngine.java:112)
	at controllers.TestRunner.run(TestRunner.java:66)
	at sun.reflect.GeneratedMethodAccessor50.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
	at play.server.PlayHandler$NettyInvocation.execute(PlayHandler.java:257)
	at play.Invoker$Invocation.run(Invoker.java:278)
	at play.server.PlayHandler$NettyInvocation.run(PlayHandler.java:235)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:98)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:206)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
	at java.lang.Thread.run(Thread.java:680)

          </system-err>  
        }
            </testcase>
  <testcase classname="ApplicationTest" name="updateUserFromJson" time="0.066">
                        <error type="" message="A java.lang.RuntimeException has been caught, java.util.concurrent.ExecutionException: play.exceptions.UnexpectedException: Unexpected Error">
        <![CDATA[
          In /test/ApplicationTest.java, line 144
          
              	    Response response = PUT(&quot;/user/&quot; + user.id, &quot;application/json&quot;, json); :
          
        ]]>
        </error>
                  <system-err>
            java.lang.RuntimeException: java.util.concurrent.ExecutionException: play.exceptions.UnexpectedException: Unexpected Error
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:299)
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:305)
	at play.test.FunctionalTest.PUT(FunctionalTest.java:231)
	at play.test.FunctionalTest.PUT(FunctionalTest.java:204)
	at ApplicationTest.updateUserFromJson(ApplicationTest.java:144)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at play.test.PlayJUnitRunner$StartPlay$2$1.evaluate(PlayJUnitRunner.java:105)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:76)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at play.test.PlayJUnitRunner.run(PlayJUnitRunner.java:55)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:24)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:157)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:117)
	at play.test.TestEngine.run(TestEngine.java:112)
	at controllers.TestRunner.run(TestRunner.java:66)
	at sun.reflect.GeneratedMethodAccessor50.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
	at play.server.PlayHandler$NettyInvocation.execute(PlayHandler.java:257)
	at play.Invoker$Invocation.run(Invoker.java:278)
	at play.server.PlayHandler$NettyInvocation.run(PlayHandler.java:235)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:98)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:206)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
	at java.lang.Thread.run(Thread.java:680)
Caused by: java.util.concurrent.ExecutionException: play.exceptions.UnexpectedException: Unexpected Error
	at java.util.concurrent.FutureTask$Sync.innerGet(FutureTask.java:232)
	at java.util.concurrent.FutureTask.get(FutureTask.java:91)
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:286)
	... 56 more
Caused by: play.exceptions.UnexpectedException: Unexpected Error
	at play.Invoker$Invocation.onException(Invoker.java:244)
	at play.Invoker$Invocation.run(Invoker.java:286)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	... 3 more
Caused by: play.mvc.results.NotFound: user.182 action not found

          </system-err>  
        }
            </testcase>
  <testcase classname="ApplicationTest" name="createBookingFromJson" time="0.209">
                        <error type="" message="A java.lang.NullPointerException has been caught, null">
        <![CDATA[
          In /test/ApplicationTest.java, line 178
          
              	    Booking newBooking = ((BookingDTO) message.content).toBooking(); :
          
        ]]>
        </error>
                  <system-err>
            java.lang.NullPointerException
	at ApplicationTest.createBookingFromJson(ApplicationTest.java:178)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at play.test.PlayJUnitRunner$StartPlay$2$1.evaluate(PlayJUnitRunner.java:105)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:76)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at play.test.PlayJUnitRunner.run(PlayJUnitRunner.java:55)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:24)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:157)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:117)
	at play.test.TestEngine.run(TestEngine.java:112)
	at controllers.TestRunner.run(TestRunner.java:66)
	at sun.reflect.GeneratedMethodAccessor50.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
	at play.server.PlayHandler$NettyInvocation.execute(PlayHandler.java:257)
	at play.Invoker$Invocation.run(Invoker.java:278)
	at play.server.PlayHandler$NettyInvocation.run(PlayHandler.java:235)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:98)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:206)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
	at java.lang.Thread.run(Thread.java:680)

          </system-err>  
        }
            </testcase>
  <testcase classname="ApplicationTest" name="createNotValidBookingFromJson" time="0.082">
      </testcase>
  <testcase classname="ApplicationTest" name="listBookingsByUser" time="0.113">
                        <error type="" message="A java.lang.RuntimeException has been caught, java.util.concurrent.ExecutionException: play.exceptions.UnexpectedException: Unexpected Error">
        <![CDATA[
          In /test/ApplicationTest.java, line 248
          
              	    response = GET(&quot;/user/&quot; + user.id + &quot;/bookings&quot;); :
          
        ]]>
        </error>
                  <system-err>
            java.lang.RuntimeException: java.util.concurrent.ExecutionException: play.exceptions.UnexpectedException: Unexpected Error
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:299)
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:305)
	at play.test.FunctionalTest.GET(FunctionalTest.java:103)
	at play.test.FunctionalTest.GET(FunctionalTest.java:57)
	at ApplicationTest.listBookingsByUser(ApplicationTest.java:248)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at play.test.PlayJUnitRunner$StartPlay$2$1.evaluate(PlayJUnitRunner.java:105)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:76)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at play.test.PlayJUnitRunner.run(PlayJUnitRunner.java:55)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:24)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:157)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:117)
	at play.test.TestEngine.run(TestEngine.java:112)
	at controllers.TestRunner.run(TestRunner.java:66)
	at sun.reflect.GeneratedMethodAccessor50.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
	at play.server.PlayHandler$NettyInvocation.execute(PlayHandler.java:257)
	at play.Invoker$Invocation.run(Invoker.java:278)
	at play.server.PlayHandler$NettyInvocation.run(PlayHandler.java:235)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:98)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:206)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
	at java.lang.Thread.run(Thread.java:680)
Caused by: java.util.concurrent.ExecutionException: play.exceptions.UnexpectedException: Unexpected Error
	at java.util.concurrent.FutureTask$Sync.innerGet(FutureTask.java:232)
	at java.util.concurrent.FutureTask.get(FutureTask.java:91)
	at play.test.FunctionalTest.makeRequest(FunctionalTest.java:286)
	... 56 more
Caused by: play.exceptions.UnexpectedException: Unexpected Error
	at play.Invoker$Invocation.onException(Invoker.java:244)
	at play.Invoker$Invocation.run(Invoker.java:286)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	... 3 more
Caused by: play.mvc.results.NotFound: GET /user/195/bookings

          </system-err>  
        }
            </testcase>
</testsuite>
